cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/FlowletProgramRunner.java

```diff
@@ -541,8 +542,10 @@ private ProcessSpecificationFactory processSpecificationFactory(
               && (inputNames.contains(queueName.getSimpleName())
               || inputNames.contains(FlowletDefinition.ANY_INPUT))) {
 
-              if (entry.getKey().getType() == FlowletConnection.Type.STREAM) {
-                ConsumerSupplier<StreamConsumer> consumerSupplier = ConsumerSupplier.create(program.getNamespace(),
+              Node sourceNode = entry.getKey();
+              if (sourceNode.getType() == FlowletConnection.Type.STREAM) {
+                 NamespaceId sourceNamespace = new NamespaceId(sourceNode.getNamespace());
```

Comment:
The sourceNode.getNamespace() will give you null here if the stream was in the same namespace from the  code changes which you have done and you pass this to ConsumerSupplier.create and i dont see changes in this PR to the  class to handle this situation. In its current implementation ConsumerSupplier.create cannot handle null as namespace and default to program namespace so here you should handle it  and pass the appropriate namespace which will be program namespace if source namespace is null.